### PR TITLE
docs: improve Array Callable Syntax explanation in routing.rst

### PR DESCRIPTION
**Pull request by kenjis**
**Description**
See #7546 
https://forum.codeigniter.com/showthread.php?tid=87864
https://forum.codeigniter.com/showthread.php?tid=87549

**Checklist:**
- [x] Securely signed commits
- [] Component(s) with PHPDoc blocks, only if necessary or adds value
- [] Unit testing, with >80% coverage
- [x] User guide updated
- [] Conforms to style guide

### CODE REVIEW

File: user_guide_src/source/incoming/routing.rst [hunk 1]
```diff
@@ -85,7 +85,8 @@ Specifying Route Handlers
 Controller's Namespace
 ----------------------
 
-If a controller name is stated without beginning with ``\``, the :ref:`routing-default-namespace` will be prepended:
+When you specify a controller and method name as a string,
+if a controller name is stated without beginning with ``\``, the :ref:`routing-default-namespace` will be prepended:
```
Comment (paulbalandan) [inline user_guide_src/source/incoming/routing.rst:89]: I do not understand what is "stated without beginning". Well, I understand this means not fully qualified but the wordings are not clear enough.

Comment (kenjis) [reply user_guide_src/source/incoming/routing.rst:89]: I want to say 
- `App\Controllers\Foo` (without beginning with ``\``) is treated as *default namespace* (`App\Controllers`) + `App\Controllers\Foo` (= `\App\Controllers\App\Controllers\Foo`), 
- but `\App\Controllers\Foo` is treated as ``\App\Controllers\Foo``

Comment (paulbalandan) [reply user_guide_src/source/incoming/routing.rst:89]: Maybe "if a controller is written without a leading \``\\``"?

File: user_guide_src/source/incoming/routing.rst [hunk 2]
```diff
@@ -114,6 +115,17 @@ Or using ``use`` keyword:
 .. literalinclude:: routing/014.php
    :lines: 2-
 
+If you forget ``use App\Controllers\Home;``, the controller classname is interpreted
```
[the remark this answers is not in the document]
Comment (kenjis) [reply user_guide_src/source/incoming/routing.rst]: Done.

**Review by paulbalandan — APPROVED** [review]

**Pull request merged**